components/mqtt.rst

```diff
@@ -137,10 +139,93 @@ Configuration options:
 -  **retain** (*Optional*, boolean): If the published message should
    have a retain flag on or not. Defaults to ``true``.
 
-.. _mqtt-using_with_home_assistant:
 
-Using with Home Assistant
--------------------------
+.. _mqtt-device_discovery:
+
+MQTT device discovery
+---------------------
+
+The ESPHome device will respond to the following MQTT topics if `mqtt.discover_ip` is enabled.
+
+- `esphome/discover` (All ESPHome device will answer)
+- `esphome/ping/<APP_NAME>`
+
+The response will be send to `esphome/discover/<APP_NAME>` and is a JSON encoded message.
+
+The MQTT device discovery is currently used for:
+
+- ESPHome dashboard (online / offline status)
+- ESPHome CLI (IP discovery e.g. for upload, logs)
+- Home Assistant device discovery
+
+Example Payload:
+
+.. code-block:: json
+
+    {
+      "ip": "192.168.0.122",
+      "name": "esp32-test",
+      "friendly_name": "Test Device",
+      "port": 6053,
+      "version": "2024.4.1",
+      "mac": "84fce6123456",
+      "platform": "ESP32",
+      "board": "esp32-c3-devkitm-1",
+      "network": "wifi",
+      "api_encryption": "Noise_NNpsk0_25519_ChaChaPoly_SHA256"
+    }
+
+
+JSON keys:
+
+-  **ip** (**Required**, ip): The IP address of the ESPHome device.
+-  **name** (**Required**, string): Name of the device (`esphome.name`).
```

Comment:
```suggestion
-  **name** (**Required**, string): Name of the device (for example: `my-device.local`).
```

Reply:
`esphome.name` is the name of the key in the yaml of the device, not the mDNS name ;)